File: Lua/1041.lua
x = io.read("*n")
y = io.read("*n")

if (x == 0 and y == 0) then
	print("Origem")
elseif (x == 0) then
	print("Eixo Y")
elseif (y == 0) then
	print("Eixo X")
elseif (x > 0 and y > 0) then
	print("Q1")
elseif (x > 0 and y < 0) then
	print("Q4")
elseif (x < 0 and y > 0) then
	print("Q2")
elseif (x < 0 and y < 0) then
	print("Q3")
end

File: Ruby/1041.rb
ponto = gets.chomp.split().map{ |x| x.to_f}
if (ponto[0] == 0 and ponto[1] == 0) then
    puts ("Origem")
elsif (ponto[0] == 0) then
    puts("Eixo Y")
elsif (ponto[1] == 0) then
    puts ("Eixo X")
elsif (ponto[0] > 0 and ponto[1] > 0) then
    puts ("Q1")
elsif (ponto[0] > 0 and ponto[1] < 0) then
    puts ("Q4")
elsif (ponto[0] < 0 and ponto[1] > 0) then
    puts ("Q2")
elsif (ponto[0] < 0 and ponto[1] < 0) then
    puts ("Q3")
end
